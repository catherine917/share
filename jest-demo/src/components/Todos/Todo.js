import React from "react";

const Todo = ({ completed, text, changeTodoState, index }) => {
  const onCheck = function(e) {
    changeTodoState(index, !completed);
    e.preventDefault();
  };
  return (
    <li className={completed ? "completed" : ""}>
      <div className="todo-item">
        <input type="checkbox" onClick={onCheck} />
        <label>{text}</label>
      </div>
    </li>
  );
};
export default Todo;
