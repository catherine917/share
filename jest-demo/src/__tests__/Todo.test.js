import React from "react";
import Todo from "../components/Todos/Todo";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("Todo", () => {
  let props, wrapper;
  beforeEach(() => {
    props = {
      index: 0,
      text: "todo-1",
      completed: false,
      changeTodoState: jest.fn(),
    };
    wrapper = shallow(<Todo {...props} />);
  });
  afterEach(() => {
      wrapper.unmount();
  })
  // 通过 input 是否存在来判断 Todo组件是否被渲染
  it("Todo item should  render", () => {
    expect(wrapper.find("input").exists());
  });

  // 当点击 单选按钮，onClick 方法应该被调用
  it("click checkbox input, onClick called", () => {
    const mockEvent = {
      key: "Click",
      preventDefault: jest.fn()
    };
    wrapper.find("input").simulate("click", mockEvent);
    expect(props.changeTodoState).toBeCalled();
  });

  // todos snapshot
  it("todos snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("done snapshot", () => {
    props = Object.assign(props,{
      completed: true
    });
    wrapper = shallow(<Todo {...props}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  })
});
