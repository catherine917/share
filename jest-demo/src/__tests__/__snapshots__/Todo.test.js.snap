// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo done snapshot 1`] = `
<li
  className="completed"
>
  <div
    className="todo-item"
  >
    <input
      onClick={[Function]}
      type="checkbox"
    />
    <label>
      todo-1
    </label>
  </div>
</li>
`;

exports[`Todo todos snapshot 1`] = `
<li
  className=""
>
  <div
    className="todo-item"
  >
    <input
      onClick={[Function]}
      type="checkbox"
    />
    <label>
      todo-1
    </label>
  </div>
</li>
`;
